ts/components/LabelledItem.tsx

```diff
@@ -187,8 +212,27 @@ export const LabelledItem: React.FC<Props> = (props: Props) => {
               onFocus={handleOnFocus}
               onBlur={handleOnBlur}
               testID={`${props.testID}Input`}
+              disabled={props.inputProps.disabled}
             />
           )}
+
+          {props.icon &&
+            iconPosition === "right" &&
+            (isString(props.icon) ? (
+              <IconFont
+                size={variables.iconSize3}
+                color={
+                  props.iconColor ? props.iconColor : variables.brandDarkGray
+                }
+                name={props.icon}
+                style={props.iconStyle}
+                onPress={props.onPressIcon}
+                accessible
+                accessibilityLabel={props.accessibilityLabelIcon}
+              />
+            ) : (
+              <Image source={props.icon} style={props.iconStyle} />
+            ))}
```

Comment:
This new block is exactly the same as [the one above](https://github.com/pagopa/io-app/pull/3143/files#diff-083d82886f4ff90d0282670bedfa8b9244dc5acfb660f0905afb9ae78f23e673R173).
Please consider extracting it to a subcomponent visible only to this module named `Icon` which takes only the properties required for the icon to be rendered and is called here.
This will reduce the cognitive complexity and hopefully let you remove [the linter exception](https://github.com/pagopa/io-app/pull/3143/files#diff-083d82886f4ff90d0282670bedfa8b9244dc5acfb660f0905afb9ae78f23e673R86)

Reply:
@pietro909 Done, I also refactored the input because the linter exception didn't go away